gpii/node_modules/flowManager/src/PSPChannel.js

```diff
@@ -139,15 +141,71 @@ gpii.pspChannel.generateDefaultSettingControls = function (defaultSettings, sche
     // Set the setting default values from reset to standard file to schema.default field for each setting in
     // "settingControls" block.
     fluid.each(defaultPreferences, function (defaultPrefsVal, defaultPrefsKey) {
-        if (schemas[defaultPrefsKey]) {
-            var schemaWithDefault = fluid.copy(schemas[defaultPrefsKey]);
-            fluid.set(schemaWithDefault, ["schema", "default"], defaultPrefsVal);
-            gpii.pspChannel.emitSettingControl(defaultSettingControls, schemaWithDefault, undefined, [defaultPrefsKey], "live");
+        var prefsInfo = gpii.pspChannel.getPreferenceInfo(schemas, defaultPrefsKey, defaultPrefsVal);
+        var schema = prefsInfo.schema;
+        if (schema) {
+            fluid.set(schema, ["default"], prefsInfo.prefsValue);
+            gpii.pspChannel.emitSettingControl(defaultSettingControls, schema, undefined, prefsInfo.prefsKeySegs, "live");
         }
     });
     return defaultSettingControls;
 };
 
+gpii.pspChannel.getPreferenceInfo = function (solutionSchemas, prefsKey, prefsValue) {
```

Comment:
Nice docs please!

Reply:
Nice docs added :) Ready for another look.